src/player.cpp

```diff
@@ -9502,14 +9499,18 @@ comfort_level player::base_comfort_value( const tripoint &p ) const
             comfort += 1 + static_cast<int>( comfort_level::slightly_comfortable );
             // Note: shelled individuals can still use sleeping aids!
         } else if( vp ) {
-            if( vp.part_with_feature( "BED", true ) ) {
-                comfort += 1 + static_cast<int>( comfort_level::slightly_comfortable );
-            } else if( vp.part_with_feature( "SEAT", true ) ) {
-                comfort += 0 + static_cast<int>( comfort_level::slightly_comfortable );
-            } else {
-                // Sleeping elsewhere is uncomfortable
-                comfort -= g->m.move_cost( p );
+            vehicle *const veh = veh_pointer_or_null( vp );
```

Comment:
```suggestion
            vehicle &veh = vp->vehicle();
```
You don't need a pointer here, and you certainly don't need to call `veh_pointer_or_null` as you have already checked that `vp` does contain a value (so it will never return null anyway).